services/brig/test/integration/API/UserPendingActivation.hs

```diff
@@ -0,0 +1,347 @@
+  (HasCallStack, MonadRandom m, MonadIO m) =>
+  m (Scim.User.User SparTag, SAML.UnqualifiedNameID)
+randomScimUserWithSubject = do
+  randomScimUserWithSubjectAndRichInfo =<< liftIO (generate arbitrary)
+
+-- | See 'randomScimUser', 'randomScimUserWithSubject'.
+randomScimUserWithSubjectAndRichInfo ::
+  (HasCallStack, MonadRandom m, MonadIO m) =>
+  RichInfo ->
+  m (Scim.User.User SparTag, SAML.UnqualifiedNameID)
+randomScimUserWithSubjectAndRichInfo richInfo = do
+  suffix <- cs <$> replicateM 7 (getRandomR ('0', '9'))
+  emails <- getRandomR (0, 3) >>= \n -> replicateM n randomScimEmail
+  phones <- getRandomR (0, 3) >>= \n -> replicateM n randomScimPhone
+  -- Related, but non-trivial to re-use here: 'nextSubject'
+  (externalId, subj) <-
+    getRandomR (0, 1 :: Int) <&> \case
+      0 ->
+        ( "scimuser_extid_" <> suffix <> "@example.com",
+          either (error . show) id $
+            SAML.mkUNameIDEmail ("scimuser_extid_" <> suffix <> "@example.com")
+        )
+      1 ->
+        ( "scimuser_extid_" <> suffix,
+          SAML.mkUNameIDUnspecified ("scimuser_extid_" <> suffix)
+        )
+      _ -> error "randomScimUserWithSubject: impossible"
+  pure
+    ( (Scim.User.empty userSchemas ("scimuser_" <> suffix) (ScimUserExtra richInfo))
+        { Scim.User.displayName = Just ("ScimUser" <> suffix),
+          Scim.User.externalId = Just externalId,
+          Scim.User.emails = emails,
+          Scim.User.phoneNumbers = phones
+        },
+      subj
+    )
+
+randomScimEmail :: MonadRandom m => m Email.Email
+randomScimEmail = do
+  let typ :: Maybe Text = Nothing
+      primary :: Maybe Bool = Nothing -- TODO: where should we catch users with more than one
+      -- primary email?
+  value :: Email.EmailAddress2 <- do
+    localpart <- cs <$> replicateM 15 (getRandomR ('a', 'z'))
+    domainpart <- (<> ".com") . cs <$> replicateM 15 (getRandomR ('a', 'z'))
+    pure . Email.EmailAddress2 $ Email.unsafeEmailAddress localpart domainpart
+  pure Email.Email {..}
+
+randomScimPhone :: MonadRandom m => m Phone.Phone
+randomScimPhone = do
+  let typ :: Maybe Text = Nothing
+  value :: Maybe Text <- do
+    let mkdigits n = replicateM n (getRandomR ('0', '9'))
+    mini <- mkdigits 8
+    maxi <- mkdigits =<< getRandomR (0, 7)
+    pure $ Just (cs ('+' : mini <> maxi))
+  pure Phone.Phone {..}
+
+-- | Create a user.
+createUser ::
+  HasCallStack =>
+  Spar ->
+  ScimToken ->
+  Scim.User.User SparTag ->
+  Http (Scim.StoredUser SparTag)
+createUser spar tok user = do
+  r <-
+    createUser_ spar (Just tok) user <!! do
+      const 201 === statusCode
+  pure (responseJsonUnsafe r)
+
+-- | Create a user.
+createUser_ ::
+  Spar ->
+  -- | Authentication
+  Maybe ScimToken ->
+  -- | User data
+  Scim.User.User SparTag ->
+  -- | Spar endpoint
+  Http ResponseLBS
+createUser_ spar auth user = do
+  -- NB: we don't use 'mkEmailRandomLocalSuffix' here, because emails
+  -- shouldn't be submitted via SCIM anyway.
```

Comment:
But they *have* to be, otherwise where would the invitation emails be sent to?  Can you make this sentence a bit clearer?  Or is this something you just copied?  In that case this may be a message to myself.  :)

Reply:
It's just something I copied. git blame says it's your message :)